Automattic/jetpack
# CI: Start using changelogger

#### Changes proposed in this Pull Request:

* Rewrite Jetpack's changelog to keepachangelog.com format.
* Enable changelogger for Jetpack.
* Have the build step write an "alpha" version entry to the changelog when changelogger is in use.
* Have CI check that changelogger change entries are valid and are being added when changelogger is in use.
* Remove the "proposed changelog entry" from the PR template.

Note this won't technically start requiring using changelogger for Jetpack changes; that will happen when we configure the new "Linting / Changelogger" job as required. But since this PR removes the "proposed changelog entry" from the PR template, we'll probably want to do that soon after merging...

#### Jetpack product discussion


p9dueE-2qL-p2
1199658893173824-as-1199910496214526

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* See that the new CI job is running properly.
* Check that the build artifact shows a "9.6-alpha" version in Jetpack's CHANGELOG.md.
* Create a PR based on this one, touching files in Jetpack and/or Changelogger, without adding a change file and see that the CI job correctly fails it.
* Create a PR based on this one with a broken change file (e.g. add an invalid header) and see that the CI job correctly fails it.
